client/client.go

```diff
@@ -23,6 +23,9 @@ var ErrRedirect = errors.New("unexpected redirect in response")
 // Client is the API client that performs all operations
 // against a docker server.
 type Client struct {
+	auth     bool
```

Comment:
Why does the client need to take care of whether it's auth or not?

Reply:
Also I think a random generated name like https://github.com/tensorchord/envd/blob/main/pkg/util/fileutil/namegenerator.go is better than random token as the username

Reply:
I changed my mind. I think it's better to force users to provide a username and password, to make it a consistent behavior across auth and no-auth. It won't make any difficult for users to choose a username for themselves

Reply:
> I changed my mind. I think it's better to force users to provide a username and password, to make it a consistent behavior across auth and no-auth. It won't make any difficult for users to choose a username for themselves

client needs to know if the jwt token is needed.

Reply:
> For noauth scenario, you can still sign a JWT token to the client. Thus it will make no difference at client side

SGTM